haystack/nodes/prompt/invocation_layer/hugging_face.py

```diff
@@ -28,6 +28,11 @@
     class StopWordsCriteria(StoppingCriteria):
         """
         Stops text generation if any one of the stop words is generated.
+
+        Note: this class only examines the truly new-generated text. For LLMs, which is designed
+        for dialogue generation and outputs the newly generated text together with the prompt
+        like `mosaicml/mpt-7b-chat`, this class will stop generating after the first token.
+        In this case, you need to make sure that your prompt template doesn't contain any stop word.
         """
```

Comment:
How about the following @faaany 

When a stop word is encountered, the generation of new text is stopped. However, if the stop word is in the prompt itself, it can stop generating new text prematurely after the first token. This is particularly important for LLMs designed for dialogue generation. For these models, like for example `mosaicml/mpt-7b-chat`, the output includes both the new text and the original prompt. Therefore, it's important to make sure your prompt has no stop words.